### PR TITLE
CPM-856: Remove a condition

### PR DESCRIPTION
**Pull request by Beninho**
**Description (for Contributor and Core Developer)**

Be able to remove a condition

**Definition Of Done (for Core Developer only)**

- [X] Tests

### CODE REVIEW

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 1]
```diff
@@ -14,10 +15,12 @@ type SelectionTabProps = {
 };
 
 type ConditionsWithIdentifier = (Condition & {id: string})[];
+type ConditionId = string;
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:18]:
```suggestion
type ConditionId = string;
type ConditionsWithIdentifier = (Condition & {id: ConditionId})[];
```
If you introduce ConditionId, use it ! (And you should rename it "ConditionIdentifier" by the way)

File: components/identifier-generator/front/src/feature/tabs/conditions/__tests__/EnabledLine.test.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import {fireEvent, render, screen} from '../../../tests/test-utils';
+import {fireEvent, render, screen, waitFor} from '../../../tests/test-utils';
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/tabs/conditions/__tests__/EnabledLine.test.tsx:2]: useless import

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 3]
```diff
@@ -74,16 +96,19 @@ const SelectionTab: React.FC<SelectionTabProps> = ({target, conditions, onChange
...
+                    <EnabledLine condition={conditionWithId} key={conditionWithId.id}
+                                 onChange={handleChange} onDelete={onDelete(conditionWithId.id)}/>
                   )}
                 </>
               ))}
             </>
           )}
         </Table.Body>
       </Table>
+      {conditionIdToDelete && <DeletePropertyModal onClose={closeModal} onDelete={handleDeleteCondition} />}
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:108]: No problem to use "DeletePropertyModal" Component here as it's the same for Property and Condition.
But you should rename it with a more generic name, as it does not delete property anymore.
Something like
- SimpleDeleteModal
- DeletePropertyOrConditionModal

Don't forget to update names of the files, the mocks and the tests.

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 4]
```diff
@@ -49,6 +52,25 @@ const SelectionTab: React.FC<SelectionTabProps> = ({target, conditions, onChange
     onChange(removeIdentifiers(newConditions));
   };
 
+  const closeModal = () => {
+    setConditionIdToDelete(undefined);
+  };
+
+  const handleDeleteCondition = (): void => {
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:59]: As we seen this morning, here is a good example of a good use for `useCallback` :D

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 5]
```diff
@@ -61,7 +83,7 @@ const SelectionTab: React.FC<SelectionTabProps> = ({target, conditions, onChange
           {isLoading && <ListSkeleton />}
           {!isLoading && (
             <>
-              <Table.Row>
+              <Table.Row key={0}>
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:86]: is it really necessary? I though keys were useful inside `map` functions

**Review by cemorin — APPROVED** [review]

File: components/identifier-generator/front/src/feature/tabs/__tests__/SelectionTab.test.tsx [hunk 6]
```diff
@@ -56,4 +57,43 @@ describe('SelectionTab', () => {
...
+    );
+
+    expect(await screen.findByText('Sku')).toBeInTheDocument();
+
+    expect(screen.getByText('EnabledLineMock')).toBeInTheDocument();
+    expect(screen.getByText('Delete Enabled')).toBeInTheDocument();
+    fireEvent.click(screen.getByText('Delete Enabled'));
+    expect(screen.getByText('SimpleDeleteModalMock')).toBeInTheDocument();
+    fireEvent.click(screen.getByText('Delete property'));
+    expect(screen.queryByText('SimpleDeleteModalMock')).not.toBeInTheDocument();
```
Comment (mmetayer) [inline components/identifier-generator/front/src/feature/tabs/__tests__/SelectionTab.test.tsx:78]: \+ maybe expect(onChange).toHaveBeenCalled()...?

File: components/identifier-generator/front/src/feature/tabs/__tests__/SelectionTab.test.tsx [hunk 7]
```diff
@@ -56,4 +57,43 @@ describe('SelectionTab', () => {
...
+      <SelectionTab target={'sku'} conditions={[{type: CONDITION_NAMES.ENABLED, value: true}]} onChange={onChange} />
+    );
+
+    expect(await screen.findByText('Sku')).toBeInTheDocument();
+
+    expect(screen.getByText('EnabledLineMock')).toBeInTheDocument();
+    fireEvent.click(screen.getByText('Delete Enabled'));
+    expect(screen.getByText('SimpleDeleteModalMock')).toBeInTheDocument();
+    fireEvent.click(screen.getByText('Close modal'));
+    expect(screen.queryByText('SimpleDeleteModalMock')).not.toBeInTheDocument();
```
Comment (mmetayer) [inline components/identifier-generator/front/src/feature/tabs/__tests__/SelectionTab.test.tsx:97]: \+ expect(onChange).not.toHaveBeenCalled()... (or else it's exactly the same as the previous test)

**Review by antoinearrive — APPROVED** [review]

**Pull request merged**